File: haskell/src/Solutions/Euler099.hs
module Solutions.Euler099 where
import Utils.Input(getInput)
import Utils.List(maxBy, splitOn)

power :: String -> Integer
power str = let [a, b] = splitOn ',' str in read a ^ read b

euler099 :: IO String
euler099 = show . fst . maxBy snd . zip [1..] . map power . words <$> getInput 99

File: rust/src/solutions/euler099.rs
use crate::utils::input::get_input;
use itertools::Itertools;
use ordered_float::OrderedFloat;

pub fn euler099() -> String {
    (1..)
        .zip(get_input(99).lines().map(|line| {
            let (base_str, exp_str) = line.split(',').take(2).collect_tuple().unwrap();
            OrderedFloat(exp_str.parse::<f64>().unwrap() * base_str.parse::<f64>().unwrap().ln())
        }))
        .max_by_key(|&(_, pow)| pow)
        .unwrap()
        .0
        .to_string()
}
